sonaranalyzer-dotnet/src/SonarAnalyzer.CSharp/Rules/LdapConnectionShouldBeSecure.cs

```diff
@@ -43,14 +47,64 @@ public sealed class LdapConnectionShouldBeSecure : ObjectShouldBeInitializedCorr
             isAllowedConstantValue: IsAllowedValue,
             trackedTypes: TrackedTypes,
             isTrackedPropertyName: propertyName => "AuthenticationType" == propertyName,
+            isAllowedObject: IsAllowedObject,
             trackedConstructorArgumentIndex: 3
         );
 
         private static bool IsAllowedValue(object constantValue) =>
             constantValue is int integerValue &&
-            (integerValue & AuthenticationTypes_Secure) > 0; // The expected value is a bit from a Flags enum
-
+            IsValidAuthenticationType(integerValue);
 
         private static readonly ImmutableArray<KnownType> TrackedTypes = ImmutableArray.Create(KnownType.System_DirectoryServices_DirectoryEntry);
+
+        private static bool IsAllowedObject(ISymbol symbol, SyntaxNode node, SemanticModel semanticModel)
+        {
+            if (!symbol.GetSymbolType().Is(KnownType.System_DirectoryServices_AuthenticationTypes))
+            {
+                return false;
+            }
+
+            // In order to correctly detect all the cases we will need to refactor the rule and use symbolic execution.
+            // Until this is done, we can reduce the number of false positives by checking if the variable has one of the following
+            // values assigned in the parent scope: AuthenticationTypes.None, AuthenticationTypes.Anonymous, default or default(AuthenticationTypes).
+            var root = node.FirstAncestorOrSelf<MethodDeclarationSyntax>() ??
+                       node.FirstAncestorOrSelf<ClassDeclarationSyntax>() ??
+                       node.FirstAncestorOrSelf<StructDeclarationSyntax>() ??
+                       node.FirstAncestorOrSelf<InterfaceDeclarationSyntax>() ??
+                       node.SyntaxTree.GetRoot();
+
+            return !HasUnsafeDeclaration(root, symbol, semanticModel) &&
```

Comment:
it would be nice to add a comment specifying what these two checks do, the scenario(s), because it's not obvious and the methods themselves don't help either

and basically explain how it's complementary to what the ObjectInitializationTracker verifies